src/main/java/Jooh/parser/Parser.java

```diff
@@ -0,0 +1,220 @@
+         */
+        public static Parsed deadline(String desc, String by) {
+            return new Parsed(Type.DEADLINE, desc, by, null, null, null);
+        }
+        /**
+         * Creates a parsed representation of an 'event' command.
+         *
+         * @param desc Description of the event.
+         * @param from Event start time string.
+         * @param to   Event end time string.
+         * @return A Parsed instance of type EVENT with description and time range.
+         */
+        public static Parsed event(String desc, String from, String to) {
+            return new Parsed(Type.EVENT, desc, null, from, to, null);
+        }
+        /**
+         * Creates a parsed representation of a 'mark' command.
+         *
+         * @param i 1-based index of the task to mark as done.
+         * @return A Parsed instance of type MARK with index.
+         */
+        public static Parsed mark(int i) {
+            return new Parsed(Type.MARK, null, null, null, null, i);
+        }
+        /**
+         * Creates a parsed representation of an 'unmark' command.
+         *
+         * @param i 1-based index of the task to mark as not done.
+         * @return A Parsed instance of type UNMARK with index.
+         */
+        public static Parsed unmark(int i) {
+            return new Parsed(Type.UNMARK, null, null, null, null, i);
+        }
+        /**
+         * Creates a parsed representation of a 'delete' command.
+         *
+         * @param i 1-based index of the task to delete.
+         * @return A Parsed instance of type DELETE with index.
+         */
+        public static Parsed delete(int i) {
+            return new Parsed(Type.DELETE, null, null, null, null, i);
+        }
+        /**
+         * Creates a parsed representation of a 'find' command.
+         *
+         * @param keyword The keyword to search for in task descriptions.
+         * @return A {@code Parsed} instance of type FIND containing the keyword.
+         */
+        public static Parsed find(String keyword) {
+            return new Parsed(Type.FIND, keyword, null, null, null, null);
+        }
+        /**
+         * Parses a raw user input string into a structured Parsed object.
+         * Splits the input into command keyword and arguments, validates them,
+         * and constructs the appropriate Parsed instance.
+         *
+         * @param input Raw command line entered by the user.
+         * @return A Parsed representation of the command.
+         * @throws JoohException If the input is empty, malformed, or violates
+         *                       expected command syntax.
+         */
+        public static Parsed parse(String input) throws JoohException {
+            if (input == null) input = "";
+            input = input.trim();   //trim removes last spacing
+            if (input.isEmpty()) throw new JoohException("Please key in something...");
```

Comment:
According to convention the loop body should be wrapped by curly braces in Egyptian style even if it's only one line. Do take note!
```suggestion
            if (input == null) {
                input = "";
            }
            input = input.trim();   //trim removes last spacing
            if (input.isEmpty()) {
                throw new JoohException("Please key in something...");
            }
```